### PR TITLE
feat: markets category route

### LINKED ISSUE #7683: Markets Page : Category Page
[notion](https://www.notion.so/shapeshift/Markets-Page-Desktop-efad92c67eee47298d2e65a133e0ace8)
[figma](https://www.figma.com/design/cuSW3HCchxWeKC8CEvn029/Markets-Page-%26-Watchlist?node-id=814-2039&t=0BpTRBv2et0FfCHi-0)

- Implement the new category page as show in mock
- add route so links are shareable
- limit to 100 assets
- Card links to asset details

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box - adds a category/<category> route for a single category, with a 100 assets limit

## Issue (if applicable)

closes #7683

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

None

## Testing

- Clicking to a category title links to a single category page
- Tabs and regular page heading isn't present on category page
- More assets than the 8 on recommended page are displayed (max is 100, though given we don't support all upstream supported ChainIds e.g Solana, actual count may be less depending on categories)

### Engineering

- ^

### Operations

- [x] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)

Screenshot 2024-09-24 at 12 07 30
Screenshot 2024-09-24 at 12 07 36
Screenshot 2024-09-24 at 12 07 49
Screenshot 2024-09-24 at 12 07 56
Screenshot 2024-09-24 at 12 08 08

### CODE REVIEW

File: src/pages/Markets/Category.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import type { MARKETS_CATEGORIES } from './constants'
+import { useRows } from './hooks/useRows'
+import { MarketsHeader } from './MarketsHeader'
+
+const containerPaddingX = { base: 4, xl: 0 }
+
+const ASSETS_LIMIT = 100
+
+export const Category: React.FC = () => {
+  const params: { category: MARKETS_CATEGORIES } = useParams()
```
Comment (kaladinlight) [inline src/pages/Markets/Category.tsx:18]: nitpick suggestion
```suggestion
  const params = useParams<{ category: MARKETS_CATEGORIES } >()
```

File: src/pages/Markets/MarketsHeader.tsx [hunk 2]
```diff
@@ -38,6 +38,21 @@ export const MarketsHeader = () => {
     history.push('/explore')
   }, [history])
 
+  const maybeCategory = useMemo(
+    () =>
+      history.location.pathname.match(/\/markets\/category\/(?<category>[\w]+)/)?.groups?.category,
```
Comment (kaladinlight) [inline src/pages/Markets/MarketsHeader.tsx:43]: suggestion: history.location.pathname.split('/').pop() (just pop off the last path param)

Comment (kaladinlight) [reply src/pages/Markets/MarketsHeader.tsx:43]: nevermind, this is short sighted. all routes have a final part to the path lol. disregard.

**Review by kaladinlight — APPROVED** [review]

File: src/pages/Markets/Recommended.tsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ export const Recommended: React.FC = () => {
           subtitle={row.subtitle}
           supportedChainIds={row.supportedChainIds}
           category={row.category}
+          showSparkline={i === 0}
```
Comment (gomesalexandre) [inline src/pages/Markets/Recommended.tsx:30]: 💜 @kaladinlight !

File: src/pages/Markets/Rows.tsx
Comment (gomesalexandre) [inline src/pages/Markets/Rows.tsx:1]: also ty @kaladinlight !

**Pull request merged**